documentation/src/main/asciidoc/manual-index.asciidoc

```diff
@@ -361,3 +361,226 @@ parameters through - see <<lucene-indexing-performance>> for more details.
+        .build();
+long executionId = BatchRuntime
+        .getJobOperator()
+        .start( MassIndexingJob.NAME, jobProps );
+----
+====
+
+In the following paragraphs, we will explain why JSR-352 standard is a good choice, and how to
+configure different parameters in order to bootstrap the mass-indexing job.
+
+==== Define your indexing scope
+Our job allows you to define your own indexing scope—you can start a full indexation or a partial
+indexation through 3 different methods: selecting the desired entity types, using HQL, or using
+Hibernate criteria.
+
+.How to add define the indexing scope in different methods?
+====
+[source, JAVA]
+----
+// full indexation
+Properties jobProps1 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .build();
+
+// partial indexation using HQL
+Properties jobProps2 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( "select c from MyClass c where c.name in ( 'Foo', 'Bar' )" )
+        .build();
+
+// partial indexation using HQL
+Properties jobProps3 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( Restrictions.in( "name", "Foo", "Bar" ) )
+        .build();
+----
+====
+
+==== Parallelism
+In order to maximize the performance, we highly recommend you to speed up the mass indexer using
+parallelism. Parallelism is activated by default. Under the standard JSR 352, the exact word is
+_"partitioning"_: The indexing step may run as multiple partitions, one per thread. Each partition
+has its own partition ID and parameters. If there’re more partitions than threads, partitions are\
+considered as a queue to consume: each thread can only run one partition at a time and won’t consume
+the next partition until the previous one is finished.
+
+===== Threads
+The maximum number of threads used by the job execution is defined through method `maxThreads()`.
+Within the N threads given, there’s 1 thread reserved for the core, so only N - 1 threads are
+available for different partitions. The default number of threads used in Hibernate Search is 10,
+which correspond to the max number of threads in JBeret. You can overwrite you maximum number used
+by Hibernate Search batch job, but we cannot guarantee that your batch runtime can provide the
+requested number.
+
+====
+[source, JAVA]
+----
+MassIndexingJob.parameters()
+        .maxThreads( 5 )
+        ...
+----
+====
+
+[NOTE]
+====
+Note that the batch runtime cannot guarantee the requested number of threads are available, it will
+use as it can up to the requested maximum. (JSR352 v1.0 Final Release, page 34)
+====
+
+This approach is optimized for best performance but requires to set the application in maintenance
+mode. Running a mass indexer in busy hours is not recommended. You also need to ensure the batch
+runtime has enough available threads for the execution, because all the batch jobs share the same
+thread pool.
```

Comment:
There are two topics here:

> This approach is optimized for best performance but requires to set the application in maintenance mode. Running a mass indexer in busy hours is not recommended.

This is general advice about how to run the job, and should in my opinion go to a separate section, preferably not in the middle of technical details.

>You also need to ensure the batch runtime has enough available threads for the execution, because all the batch jobs share the same thread pool.

This seems redundant with the note just above, or at least it's dealing with a very similar problem. Maybe we should add this to the note as a second paragraph?